.github/BOTMETA.yml

```diff
@@ -1227,18 +1213,15 @@ files:
   test/integration/targets/aci:
     maintainers: $team_aci
     labels:
-    - aci
```

Comment:
This is incorrect, we are matching a substring here.

Reply:
As far as i know, botmeta handles substrings like a whole path: `aci` is a path component of `test/integration/targets/aci`. There are four path components in `test/integration/targets/aci`: `test`, `integration`, `targets` and `aci`, [`test`](https://github.com/ansible/ansible/labels?q=test) and [`aci`](https://github.com/ansible/ansible/labels?q=aci) are valid GitHub labels, only this ones are kept.

I run manually ansibullbot with this diff on 43847:
```
diff --git a/ansibullbot/parsers/botmetadata.py b/ansibullbot/parsers/botmetadata.py
index 7faa674..3dd94e4 100644
--- a/ansibullbot/parsers/botmetadata.py
+++ b/ansibullbot/parsers/botmetadata.py
@@ -67,6 +67,8 @@ class BotMetadataParser(object):
 
         def extend_labels(data):
             for k, v in data['files'].items():
+                if k == "test/integration/targets/aci":
+                    v.pop('labels')
                 # labels from path(s)
                 if v is None:
                     continue
```
[self.meta.get('component_labels')](https://github.com/ansible/ansibullbot/blob/149f8e099a8d5822805418c62c73f3fd6256371f/ansibullbot/triagers/ansible.py#L1015) is still equal to `['aci', 'test']`.

Reply:
@pilou- Hmm, alright, I would have expected it to first match the path, and only then use path-components for matching labels. Because in our case, the paths would not be *test/integration/targets/aci*, but rather *test/integration/target/aci_rest/* and others... So I was expecting it to match `aci_rest`, and not `aci`.

PS That's why I prefer to use a trailing slash if it's known that the path is a directory and not a substring match.

Reply:
So should I update these entries keeping labels and adding an underscore (for example: `test/integration/target/aci_`) ?

Reply:
That's a possibility, but I don't know if this is the same as was intended. (I.e. it is possible that the targets do not reflect modules per se). I'd rather add trailing slashes in those cases where it is a directory and leave it as is. Thanks for clarifying on how Ansibot works ;-)

Reply:
Ok, rebased :)